core/server/master/src/main/java/alluxio/master/file/meta/ExtendedAttribute.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+/**
+ * This enum is used to define constant values for extended attribute keys on inodes within
+ * the inode tree.
+ *
+ *
+ *
+ * Extended attributes take the form of
+ * {@code {NAMESPACE}{SEPARATOR}{ATTRIBUTE_name}}
+ *
+ * For example, the persis
+ */
+public enum ExtendedAttribute {
+
+  PERSISTENCE_STATE("ps"), // system (ufs) property
+  PIN_STATE(NamespacePrefix.USER, "pin"), // property set by user
+  ;
+
+  private static final String SEPARATOR = ".";
+
+  private final String mName;
+
```

Comment:
Can we include encoding / decoding information so we can automate the value conversion?

Reply:
Actually, that's not a bad idea. Instead of an enum, maybe I can create an `ExtendedAttribute` interface where each implementation defines the identifier, namespace, and encode/decode functions

Reply:
Yeah that sounds good to me.